[9962] Post level RRM settings

ankitrox opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #9962 

## Relevant technical choices

<!-- Please describe your changes. -->

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.
- [x] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

techanvil commented on assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-block-editor.scss:
There's no styling here, and this is how the dropdown label looks on my install (running WP 6.6.1):

![image](https://github.com/user-attachments/assets/31521c26-ef23-433b-98d9-5094ecdbfbb1)

The label should not be all-caps, and the text should should wrap to the next line as needed.

Let's also move the SCSS file to `blocks/reader-revenue-manager/block-editor-plugin/editor-styles.scss` as discussed below.

techanvil commented on blocks/reader-revenue-manager/components/SettingPanel.js:
```diff
@@ -0,0 +1,50 @@
+
+export default function SettingPanel() {
+	return (
+		<Fragment>
```
This `Fragment` is redundant and can be removed.

techanvil commented on blocks/reader-revenue-manager/components/SettingsForm.js:
```diff
@@ -0,0 +1,115 @@
+	}
+
+	return (
+		<Fragment>
```
This `Fragment` can also be removed.

techanvil commented on blocks/reader-revenue-manager/plugin-registration.js:
```diff
@@ -0,0 +1,55 @@
+
+const { select, resolveSelect } = Data;
+
+registerStore( Data );
+
```
It's a bit odd having this file exporting a function to register the plugin on demand, but at the same time having the unconditional call to `registerStore( Data )`.

Let's move the call to `registerStore` to `index.js` where the exported function is called. Then, in the test for this file, call `registerStore( Data )` in the test setup to ensure the tests still work.

techanvil commented on blocks/reader-revenue-manager/plugin-registration.test.js:
```diff
@@ -0,0 +1,84 @@
+	} );
+
+	it( 'should register the plugin if the module is connected', async () => {
+		dispatch( CORE_MODULES ).receiveGetModules( [
+			{
+				slug: 'reader-revenue-manager',
+				active: true,
+				connected: true,
+				shareable: true,
+				recoverable: true,
+				internal: false,
+			},
+		] );
```
Let's use `provideModules()` here, with a similar call to how it's used elsewhere for RRM:

```suggestion
		provideModules( Data, [
			{
				slug: 'reader-revenue-manager',
				active: true,
				connected: true,
			},
		] );
```

techanvil commented on blocks/reader-revenue-manager/plugin-registration.test.js:
```diff
@@ -0,0 +1,84 @@
+	} );
+
+	it( 'should not register the plugin if the module is not connected', async () => {
+		dispatch( CORE_MODULES ).receiveGetModules( [] );
```
As above. You might also want to add a test for the case where the module is not active.

```suggestion
		provideModules( Data, [
			{
				slug: 'reader-revenue-manager',
				active: true,
				connected: false,
			},
		] );
```

techanvil commented on blocks/reader-revenue-manager/index.js:
Taking a look at what's coming in https://github.com/google/site-kit-wp/pull/10247, I'd suggest a reorganisation of the files under `blocks` as follows:

```
blocks/reader-revenue-manager
├── block-editor-plugin
│   ├── editor-styles.scss
│   ├── index.js
│   ├── plugin-registration.js
│   ├── plugin-registration.test.js
│   ├── SettingPanel.js
│   └── SettingsForm.js
└── common
    └── constants.js
```

techanvil commented on blocks/reader-revenue-manager/components/SettingsForm.js:
```diff
@@ -0,0 +1,115 @@
+import { MODULES_READER_REVENUE_MANAGER } from '../../../assets/js/modules/reader-revenue-manager/datastore/constants';
+
+const { select, dispatch } = Data;
+const CORE_EDITOR = 'core/editor';
```
Let's extract this constant to `blocks/reader-revenue-manager/common/constants.js`, in line with the changes that are coming in https://github.com/google/site-kit-wp/pull/10247/files#diff-eebc7bb1047fb0c0e7a220332932857840c7bfdba8549a67ab518a829497024f.

techanvil commented on includes/Modules/Reader_Revenue_Manager.php:
```diff
@@ -449,9 +450,23 @@ protected function setup_assets() {
+				'googlesitekit-reader-revenue-manager-block-editor-plugin',
 				array(
-					'src'           => $base_url . 'js/blocks/googlesitekit-reader-revenue-manager-block-editor.js',
+					'src'           => $this->context->url( 'dist/assets/js/blocks/reader-revenue-manager/index.js' ),
```
Let's stick with using `$base_url` for consistency:

```suggestion
					'src'           => $base_url . 'js/blocks/reader-revenue-manager/index.js',
```

techanvil commented on webpack/gutenbergBlocks.config.js:
```diff
@@ -40,9 +40,9 @@ const {
-		'googlesitekit-reader-revenue-manager-block-editor':
-			'./assets/js/googlesitekit-reader-revenue-manager-block-editor.js',
+		// Reader Revenue Manager
+		'reader-revenue-manager/index':
+			'./blocks/reader-revenue-manager/index.js',
```
Again, taking future changes into account let's update this:

```suggestion
		'reader-revenue-manager/block-editor-plugin/index':
			'./blocks/reader-revenue-manager/block-editor-plugin/index.js',
```

techanvil commented on webpack/adminCss.config.js:
```diff
@@ -35,6 +35,7 @@ module.exports = ( mode ) => ( {
 		'googlesitekit-wp-dashboard-css': './assets/sass/wpdashboard.scss',
 		'googlesitekit-authorize-application-css':
 			'./assets/sass/authorize-application.scss',
+		'googlesitekit-block-editor-css': './assets/sass/block-editor.scss',
```
Let's move this to `webpack/gutenbergBlocks.config.js` similar to the approach taken in https://github.com/google/site-kit-wp/pull/10247 and https://github.com/google/site-kit-wp/pull/10191.

Let's also move the SCSS file to `blocks/reader-revenue-manager/block-editor-plugin/editor-styles.scss`.

techanvil reviewed (changes requested):
Thanks @ankitrox, this is looking good, but could use a few amendments. Please see my comments for details.

ankitrox commented:
Thank you @techanvil for reviewing the PR and adding your feedback.

I have addressed the feedback and pushed the changes.

Assigning this to you for another review.

Thanks

techanvil commented on blocks/reader-revenue-manager/block-editor-plugin/editor-styles.scss:
```diff
@@ -16,7 +16,14 @@
+		.googlesitekit-rrm-panel__select-control {
+			label {
+				text-transform: none;
+				white-space: break-spaces;
```
Let's go with `normal`, there's no need to preserve multiple spaces in the way `break-spaces` does. I'll apply this change myself to save the round trip.

techanvil commented on blocks/reader-revenue-manager/block-editor-plugin/editor-styles.scss:
```diff
@@ -16,7 +16,14 @@
- * Block editor styles.
- */
-@import "components/reader-revenue-manager/googlesitekit-rrm-block-editor";
+.editor-sidebar {
```
It looks like the `editor-sidebar` class is a [recent addition](https://github.com/WordPress/gutenberg/pull/60815/files#diff-de4b93789d100c3f45f34a01926c40c39206d34cce26efc3eb217e58becc91e3) and using it won't be backward compatible to our supported range of WP versions (from 5.4+ for this particular feature as per the AC, rather than 5.2 which is our minimum version for the plugin in general).

Please use a class which will be backward compatible to WP 5.4; if you are referencing a class that is provided by Gutenberg then it should be backward compatible to Gutenberg 6.6.0 as per https://developer.wordpress.org/block-editor/contributors/versions-in-wordpress/.

ankitrox replied:
Thanks @techanvil , I think we can safely proceed with `.block-editor__container` because this class exists in WP 5.4 as well as in the latest version. Please let me know your thoughts.

techanvil replied:
Looks good, thanks @ankitrox!

techanvil commented on blocks/reader-revenue-manager/block-editor-plugin/plugin-registration.js:
```diff
@@ -0,0 +1,52 @@
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_MODULES } from '../../../assets/js/googlesitekit/modules/datastore/constants';
+import SettingPanel from './SettingPanel';
+import { MODULES_READER_REVENUE_MANAGER } from '../../../assets/js/modules/reader-revenue-manager/datastore/constants';
```
A very minor one, but please reorder these imports for better organisation:

```suggestion
import { CORE_MODULES } from '../../../assets/js/googlesitekit/modules/datastore/constants';
import { MODULES_READER_REVENUE_MANAGER } from '../../../assets/js/modules/reader-revenue-manager/datastore/constants';
import SettingPanel from './SettingPanel';
```

techanvil commented on blocks/reader-revenue-manager/block-editor-plugin/SettingPanel.js:
```diff
@@ -0,0 +1,46 @@
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+// eslint-disable-next-line import/no-unresolved
```
Hey @ankitrox, one more thing - we no longer to disable `import/no-unresolved` on a per-line basis, as this is handled globally via ESLint config:

https://github.com/google/site-kit-wp/blob/aadae534f6f152ca837ae4e643d3bdfb73e7b5eb/.eslintrc.json#L17-L24

techanvil reviewed (approved):
Thanks @ankitrox! This is LGTM.

:white_check_mark: 

Note that I've confirmed the failing E2E test is unrelated to the changes in this PR.
